### PR TITLE
DM-18000: Implement PipelineTask and yaml file to convert DiaSources for SDM system inside ap_association

### PR DESCRIPTION
**Pull request by morriscb**

### CODE REVIEW

File: python/lsst/ap/association/transformDiaSourceCatalog.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+            funcs = self.getFunctors()
+        analysis = PostprocessAnalysis(parq, funcs, filt=band)
+        return analysis
+
+    def getFunctors(self):
+        funcs = CompositeFunctor.from_file(self.config.functorFile)
+        funcs.update(dict(PostprocessAnalysis._defaultFuncs))
+        return funcs
+
+    # Below are temporary functions to preserve functionality before
```
Comment (timothydmorton) [inline python/lsst/ap/association/transformDiaSourceCatalog.py]: I'm assuming this comment means the methods below are used in some other code somewhere for the time being?  If not, they should probably be removed.

**Review by timothydmorton — APPROVED** [review]

File: python/lsst/ap/association/transformDiaSourceCatalog.py (hunk 1 above)
Comment (morriscb) [reply python/lsst/ap/association/transformDiaSourceCatalog.py]: Correct. You just caught me before I got rid of them. Check the compare again and you'll see that this and `getFunctors` is now gone and inherited from the `TransformBaseCatalogTask` in postprocess.py in `pipe_tasks`.

File: python/lsst/ap/association/transformDiaSourceCatalog.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    def getAnalysis(self, parq, funcs=None, band=None):
+        # Avoids disk access if funcs is passed
+        if funcs is None:
+            funcs = self.getFunctors()
+        analysis = PostprocessAnalysis(parq, funcs, filt=band)
+        return analysis
+
+    def getFunctors(self):
+        funcs = CompositeFunctor.from_file(self.config.functorFile)
+        funcs.update(dict(PostprocessAnalysis._defaultFuncs))
```
[the remark this answers is not in the document]
Comment (morriscb) [reply python/lsst/ap/association/transformDiaSourceCatalog.py]: See my reply a few comments down.

File: python/lsst/ap/association/transformDiaSourceCatalog.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+        diaSourceDf = diaSourceCat.asAstropy().to_pandas()
+        diaSourceDf["bboxSize"] = self.computeBBoxSizes(diaSourceCat)
+        diaSourceDf["ccdVisitId"] = ccdVisitId
+        diaSourceDf["filterName"] = band
+        diaSourceDf["midPointTai"] = diffIm.getInfo().getVisitInfo().getDate().get(system=DateTime.MJD)
+        diaSourceDf["diaObjectId"] = 0
+        diaSourceDf["htmId20"] = 0
+
+        df = self.transform(band,
```
[the remark this answers is not in the document]
Comment (morriscb) [reply python/lsst/ap/association/transformDiaSourceCatalog.py:140]: Analysis in this context is a `PostprocessAnalysis` (https://github.com/lsst/pipe_tasks/blob/0a6ae7924aec43e43d9a5ca4990e8100581c9634/python/lsst/pipe/tasks/postprocess.py#L341) object that is used by the `TransformBaseCatalog` and inherited classes to run the functors. I don't see a reason for returning it as it's a class to run the functors and clean up if requested.

**Pull request merged**